src/analyzer/schema-generator.ts

```diff
@@ -1238,7 +1242,7 @@ function extractLowestCommonAncestorFolderFromModels<T extends PageModel | DataM
 }
 
 function findLowestCommonAncestorFolder(filePaths: string[]): string {
-    let commonDir = path.parse(filePaths[0]!).dir;
+    let commonDir = path.parse(filePaths[0] ?? '').dir;
```

Comment:
@denar90 in what scenario the `filePath` is an empty array?
What function can call it with empty array?

Reply:
this one https://github.com/stackbit/stackbit-sdk/blob/23868b15af25dcea238462654d7b3fcc2dbbff2a/src/analyzer/schema-generator.ts#L209

Reply:
Then this is the "**Root Cause**" of the problem, not where you fixed.
Just make sure before this line to check if pageModels are empty and early return.

Reply:
If there is no files at all, then it doesn't make sense even returning empty string because for no files, there is no common ancestor